File: memory-domain/src/main/java/com/memory/domain/hashtag/repository/HashTagRepositoryCustom.java
package com.memory.domain.hashtag.repository;

import com.memory.domain.hashtag.HashTag;

import java.util.List;
import java.util.Optional;

public interface HashTagRepositoryCustom {
    List<HashTag> findPopularHashTags(int limit);
    List<HashTag> findHashTagsByNameContaining(String keyword, int limit);
    Optional<HashTag> findByName(String name);
}

File: memory-domain/src/main/kotlin/com/memory/domain/hashtag/repository/HashTagRepositoryCustom.kt
package com.memory.domain.hashtag.repository

import com.memory.domain.hashtag.HashTag
import java.util.Optional

interface HashTagRepositoryCustom {
    fun findPopularHashTags(limit: Int): List<HashTag>
    fun findHashTagsByNameContaining(keyword: String?, limit: Int): List<HashTag>
    fun findByName(name: String?): Optional<HashTag>
}
